pkg/deploy/deploy_test.go

```diff
@@ -747,7 +748,9 @@ func TestDeployConfigsWithDeploymentErrors(t *testing.T) {
 		deploy.EnvironmentInfo{Name: env}: deploy.DummyClientSet,
 	}
 
-	t.Run("deployment error - stop on error", func(t *testing.T) {
+	t.Run("[non-parallel] deployment error - stop on error", func(t *testing.T) {
```

Comment:
FYI @Laubi I've replaced your original "skip test if flag is on" version with one that explicitly setup the flags for the tests. 
I think that makes the desired behaviour clearer, and ensures all  behaviours are tested instead of basically always skipping the continue-test. 

It should also make it obvious which tests to remove when we drop the non parallel deployments eventually